torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -68,6 +68,7 @@ def create_runtime_wrapper(
     trace_joint: bool,
     keep_input_mutations: bool,
     disable_amp: bool,
+    fakified_first_outputs: Optional[List[Tensor]] = None,
```

Comment:
bleh - today, this `create_runtime_wrapper()` is meant to be (mostly) agnostic to training vs. inference. So it's pretty confusing that the "fakified_first_outputs" runtime logic gets run in the runtime wrapper for inference, but not for training.

I think I'd rather just manually shove this logic somewhere here instead https://github.com/pytorch/pytorch/pull/120523/files#diff-cb8e02fc8f37e53904ab1b151c46dd109cf50d8121bbd340834b2e976b22ebc4R159 (so the `aot_dispatch_base` and `aot_dispatch_autograd` paths both have some shared helper fn they can call). Does that sound ok?

Reply:
Sounds good, will update